From 9e3e338e759cd31d84ba79536d138346565a2e75 Mon Sep 17 00:00:00 2001
From: cgns Upstream <kwrobot@kitware.com>
Date: Fri, 6 Mar 2020 13:39:38 -0500
Subject: [PATCH] cgns 2020-03-06 (23616c5b)

Code extracted from:

    https://gitlab.kitware.com/third-party/cgns.git

at commit 23616c5bdc0f052354030941d9dc78efc284573b (for/paraview-20200309-4.0.0).
---
 CMakeLists.txt       |  4 ++--
 src/cgns_internals.c |  4 ++--
 src/cgnslib.c        | 10 +++-------
 src/cgnslib.h        |  4 ++--
 4 files changed, 9 insertions(+), 13 deletions(-)

--- a/ThirdParty/cgns/vtkcgns/CMakeLists.txt-org	2020-04-06 10:12:56.802109000 -0700
+++ b/ThirdParty/cgns/vtkcgns/CMakeLists.txt	2020-04-06 10:14:18.071110604 -0700
@@ -15,11 +15,11 @@
 endif ()
 
 project("cgns" C)
-set(CGNS_VERSION "3.4")
+set(CGNS_VERSION "4.0")
 
 if (FALSE) # XXX(kitware): ParaView doesn't need CPack here.
 # Allow for building a package
-set(CPACK_PACKAGE_VERSION "${CGNS_VERSION}-develop")
+set(CPACK_PACKAGE_VERSION "${CGNS_VERSION}")
 set(CPACK_PACKAGE_INSTALL_DIRECTORY "cgns-${CGNS_VERSION}")
 set(CPACK_GENERATOR "STGZ;TGZ;ZIP")
 set(CPACK_SOURCE_GENERATOR "STGZ;TGZ;ZIP")
diff --git a/src/cgns_internals.c b/src/cgns_internals.c
index 0f47726c6c8..e32564397fa 100644
--- a/ThirdParty/cgns/vtkcgns/src/cgns_internals.c
+++ b/ThirdParty/cgns/vtkcgns/src/cgns_internals.c
@@ -1151,7 +1151,7 @@ int cgi_read_section(int in_link, double parent_id, int *nsections,
                         section[0][n].connect->data = (void *)elem_data;
                     }
                 }
-                if (cg->version < 3400) {
+                if (cg->version < 4000) {
                     cgsize_t size, *elem_data = 0;
                     if (section[0][n].el_type == CGNS_ENUMV(NGON_n) ||
                         section[0][n].el_type == CGNS_ENUMV(NFACE_n) ) {
@@ -8733,7 +8733,7 @@ cgsize_t cgi_element_data_size(CGNS_ENUMT(ElementType_t) type,
         if (connect == 0) return CG_OK;
         /* Need to handle old version when opening old files */
         if (connect_offset == 0) {
-            if (cg->version < 3400) {
+            if (cg->version < 4000) {
                 for (ne = 0; ne < nelems; ne++) {
                     npe = (int)connect[size++];
                     size += npe;
diff --git a/src/cgnslib.c b/src/cgnslib.c
index 3d6ead26eb1..7b4110b0657 100644
--- a/ThirdParty/cgns/vtkcgns/src/cgnslib.c
+++ b/ThirdParty/cgns/vtkcgns/src/cgnslib.c
@@ -3041,15 +3041,11 @@ int cg_ElementPartialSize(int file_number, int B, int Z, int S,
         return CG_OK;
     }
 
-    if (read_element_data(section)) return CG_ERROR;
     if (read_offset_data(section)) return CG_ERROR;
-
-    data = (cgsize_t *)section->connect->data;
     offset_data = (cgsize_t *)section->connect_offset->data;
-    offset = cgi_element_data_size(section->el_type,
-                 start - section->range[0], data, offset_data);
-    if (offset < 0) return CG_ERROR;
-    size = offset_data[end] - offset_data[start-1];
+    if (offset_data == 0) return CG_ERROR;
+
+    size = offset_data[end-section->range[0]+1] - offset_data[start-section->range[0]];
     if (size < 0) return CG_ERROR;
     *ElementDataSize = size;
     return CG_OK;
diff --git a/ThirdParty/cgns/vtkcgns/src/cgnslib.h b/ThirdParty/cgns/vtkcgns/src/cgnslib.h
index cc5cbf4148c..9e4a54352df 100644
--- a/ThirdParty/cgns/vtkcgns/src/cgnslib.h
+++ b/ThirdParty/cgns/vtkcgns/src/cgnslib.h
@@ -35,8 +35,8 @@
 #ifndef CGNSLIB_H
 #define CGNSLIB_H
 
-#define CGNS_VERSION 3400
-#define CGNS_DOTVERS 3.40
+#define CGNS_VERSION 4000
+#define CGNS_DOTVERS 4.00
 
 #define CGNS_COMPATVERSION 2540
 #define CGNS_COMPATDOTVERS 2.54
